### PR TITLE
Fix for TN-453 - `resume_identifier` rather than `resume_id`

### PR DESCRIPTION
**Pull request by pbugni**
Provide the assessment engine with the actual identifier value from the QuestionnaireResponse for more reliable session behavior.

### CODE REVIEW

File: portal/models/fhir.py [hunk 1]
```diff
@@ -572,6 +572,30 @@ def aggregate_responses(instrument_ids, current_user):
...
 
+
+def qnr_document_id(
+        subject_id, questionniare_bank_id, questionnaire_name, status):
+    """Return document['identifier'] for matching QuestionnaireResponse
+
+    Using the given filter data to look for a matching QuestionnaireResponse.
+    Expecting to find exactly one, or raises NoResultFound
+
+    :return: the document identifier
```
Comment (ivan-c) [inline portal/models/fhir.py]: The identifier is a string, could you add `:rtype: str`?

File: tests/test_assessment_status.py [hunk 2]
```diff
@@ -21,14 +23,23 @@
 from tests import TestCase, TEST_USER_ID
 
 
-def mock_qr(instrument_id, status='completed', timestamp=None, qb=None):
+def mock_qr(
+        instrument_id, status='completed', timestamp=None, qb=None,
+        doc_id=None):
+    if not doc_id:
+        doc_id = randint(10000, 50000)
```
Comment (ivan-c) [inline tests/test_assessment_status.py]: Could you test with a string, since the `identifier` is one?

File: tests/test_assessment_status.py [hunk 3]
```diff
@@ -301,6 +312,25 @@ def mark_metastatic(self):
 
 
 class TestAssessmentStatus(TestQuestionnaireSetup):
+
+    def test_qnr_id(self):
+        qb = QuestionnaireBank.query.first()
+        mock_qr(
+            instrument_id='irondemog',
+            status='in-progress', qb=qb,
+            doc_id=21212)
```
Comment (ivan-c) [inline tests/test_assessment_status.py]: Could you test with a string, since the `identifier` is one?

File: portal/models/fhir.py (hunk 1 above)
Comment (pbugni) [reply portal/models/fhir.py]: This is python, we don't type things unless strictly necessary.  The code will work for either type, whatever is stored in the json.

Comment (ivan-c) [reply portal/models/fhir.py]: Fair enough, this is just more boilerplate to keep up to date when code changes. 
Any strong feelings about type-hinting? :)

**Pull request merged**